lib/node_modules/@stdlib/blas/base/srotg/test/test.assign.js

```diff
@@ -0,0 +1,130 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2023 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var tape = require( 'tape' );
+var Float32Array = require( '@stdlib/array/float32' );
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var srotg = require( './../lib/' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof srotg, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function has an arity of 5', function test( t ) {
+	t.strictEqual( srotg.assign.length, 5, 'arity of 5' );
+	t.end();
+});
+
+tape( 'the function computes a Givens plane rotation', function test( t ) {
+	var expected = [
+		[ 0.5, 1.6666666, 0.6, 0.8 ],
+		[ 0.5, 0.6, 0.8, 0.6 ],
+		[ 0.5, -1.6666666, -0.6, 0.8 ],
+		[ -0.5, -0.6, 0.8, -0.6 ],
+		[ -0.5, 1.6666666, 0.6, 0.8 ],
+		[ 0.0, 0.0, 1.0, 0.0 ],
+		[ 1.0, 1.0, 0.0, 1.0 ],
+		[ 1.0, 0.0, 1.0, 0.0 ]
+	];
+	var values = [
+		[ 0.3, 0.4 ],
+		[ 0.4, 0.3 ],
+		[ -0.3, 0.4 ],
+		[ -0.4, 0.3 ],
+		[ -0.3, -0.4 ],
+		[ 0.0, 0.0 ],
+		[ 0.0, 1.0 ],
+		[ 1.0, 0.0 ]
+	];
+	var out;
+	var i;
+
+	for ( i = 0; i < values.length; i++ ) {
+		expected[i] = new Float32Array( expected[i] );
+		out = new Float32Array( 4 );
+		srotg.assign( values[i][0], values[i][1], out, 1, 0 );
```

Comment:
You're passing in doubles. You need to wrap with `float64ToFloat32(...)` in order to "cast" a double-precision scalar to a single-precision equiv. In short, `float64ToFloat32( 0.3 ) !== 0.3`.